### PR TITLE
Rfx fixes

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

**Comment by akallabeth** [comment]
@pnowack I´ve added a few assertions to `rfx` encoder, I´m having a look at this today.

**Review by pnowack — APPROVED** [review]
Just some small things, but could not spot any issues.

File: libfreerdp/codec/rfx.c [hunk 1]
```diff
@@ -1350,12 +1399,12 @@ static size_t rfx_tile_length(RFX_TILE* tile)
...
 static BOOL rfx_write_tile(RFX_CONTEXT* context, wStream* s, RFX_TILE* tile)
 {
-	UINT32 blockLen;
-	blockLen = rfx_tile_length(tile);
+	const size_t blockLen = rfx_tile_length(tile);
 
 	if (!Stream_EnsureRemainingCapacity(s, blockLen))
 		return FALSE;
 
+	WINPR_ASSERT(tile);
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply libfreerdp/codec/rfx.c:1454]: yes, true. but too lazy to change that now :yum:

File: libfreerdp/codec/rfx.c [hunk 2]
```diff
@@ -1570,7 +1586,7 @@ RFX_MESSAGE* rfx_encode_message(RFX_CONTEXT* context, const RFX_RECT* rects, siz
 	maxTilesY = 1 + TILE_NO(extents->bottom - 1) - TILE_NO(extents->top);
 	maxNbTiles = maxTilesX * maxTilesY;
 
-	if (!(message->tiles = winpr_aligned_calloc(maxNbTiles, sizeof(RFX_TILE*), 32)))
+	if (!rfx_ensure_tiles(message, maxNbTiles))
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply libfreerdp/codec/rfx.c:1613]: we only allocate an array of pointers, so the additional `514*sizeof(void*)` should be tolerable.
we do try to allocate the desired number of tiles at the beginning, but I tend to double check stuff here as we had issues with corner cases (rounding errors, padding pixels in surfaces, ...) that were not considered at time of writing and added later on (worst case here we waste ~8kB of memory)

File: server/Windows/wf_update.c [hunk 3]
```diff
@@ -130,7 +130,7 @@ void wf_update_encode(wfInfo* wfi)
 		return;
 	}
 
-	wfi->frame_idx = wfi->rfx_context->frameIdx;
+	wfi->frame_idx = rfx_context_get_frame_idx(wfi->rfx_context);
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply server/Windows/wf_update.c:133]: ok, squashed.

File: libfreerdp/codec/rfx.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (pnowack) [reply libfreerdp/codec/rfx.c:1612]: > we only allocate an array of pointers, so the additional 514*sizeof(void*) should be tolerable.

Oh, right, indeed. Yeah, just the pointers is okay, didn't realize that this does not allocate the tiles itself.

**Pull request merged**